pkg/security/security_profile/profile/manager.go

```diff
@@ -0,0 +1,368 @@
+	"sync"
+
+	"github.com/DataDog/datadog-go/v5/statsd"
+	"github.com/hashicorp/golang-lru/v2/simplelru"
+	"go.uber.org/atomic"
+
+	"github.com/DataDog/datadog-agent/pkg/security/config"
+	"github.com/DataDog/datadog-agent/pkg/security/metrics"
+	proto "github.com/DataDog/datadog-agent/pkg/security/proto/security_profile/v1"
+	"github.com/DataDog/datadog-agent/pkg/security/resolvers/cgroup"
+	cgroupModel "github.com/DataDog/datadog-agent/pkg/security/resolvers/cgroup/model"
+	"github.com/DataDog/datadog-agent/pkg/security/seclog"
+)
+
+// SecurityProfileManager is used to manage Security Profiles
+type SecurityProfileManager struct {
+	config         *config.Config
+	statsdClient   statsd.ClientInterface
+	cgroupResolver *cgroup.Resolver
+	providers      []Provider
+
+	profilesLock sync.Mutex
+	profiles     map[cgroupModel.WorkloadSelector]*SecurityProfile
+
+	cacheLock sync.Mutex
+	cache     *simplelru.LRU[cgroupModel.WorkloadSelector, *SecurityProfile]
+	cacheHit  *atomic.Uint64
+	cacheMiss *atomic.Uint64
+}
+
+// NewSecurityProfileManager returns a new instance of SecurityProfileManager
+func NewSecurityProfileManager(config *config.Config, statsdClient statsd.ClientInterface, cgroupResolver *cgroup.Resolver) (*SecurityProfileManager, error) {
+	var providers []Provider
+
+	// instantiate directory provider
+	if len(config.SecurityProfileDir) != 0 {
+		dirProvider, err := NewDirectoryProvider(config.SecurityProfileDir, config.SecurityProfileWatchDir)
+		if err != nil {
+			return nil, fmt.Errorf("couldn't instantiate a new security profile directory provider: %w", err)
+		}
+		providers = append(providers, dirProvider)
+	}
+
+	profileCache, err := simplelru.NewLRU[cgroupModel.WorkloadSelector, *SecurityProfile](config.SBOMResolverWorkloadsCacheSize, nil)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't create security profile cache: %w", err)
+	}
+
+	m := &SecurityProfileManager{
+		config:         config,
+		statsdClient:   statsdClient,
+		providers:      providers,
+		cgroupResolver: cgroupResolver,
+		profiles:       make(map[cgroupModel.WorkloadSelector]*SecurityProfile),
+		cache:          profileCache,
+		cacheHit:       atomic.NewUint64(0),
+		cacheMiss:      atomic.NewUint64(0),
+	}
+
+	// register the manager to the provider(s)
+	for _, p := range m.providers {
+		p.SetOnNewProfileCallback(m.OnNewProfileEvent)
+	}
+	return m, nil
+}
+
+// Start runs the manager of Security Profiles
+func (m *SecurityProfileManager) Start(ctx context.Context) {
+	// start all providers
+	for _, p := range m.providers {
+		if err := p.Start(ctx); err != nil {
+			seclog.Errorf("couldn't start profile provider: %v", err)
+			return
+		}
+	}
+
+	// register the manager to the CGroup resolver
+	_ = m.cgroupResolver.RegisterListener(cgroup.WorkloadSelectorResolved, m.OnWorkloadSelectorResolvedEvent)
+	_ = m.cgroupResolver.RegisterListener(cgroup.CGroupDeleted, m.OnCGroupDeletedEvent)
+
+	seclog.Infof("security profile manager started")
+
+	for {
```

Comment:
you can remove the for loop I think

Reply:
nice catch ... bad copy paste I guess ! 🙇🏻